modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryServiceImpl.java

```diff
@@ -276,6 +299,50 @@ protected void unsetPortletResourcePermission(
 		_portletResourcePermissions.remove(resourceName);
 	}
 
+	private void _checkAccountEntryPermission(
+			long objectDefinitionId, long objectEntryId,
+			Map<String, Serializable> values)
+		throws PortalException {
+
+		ObjectDefinition objectDefinition =
+			_objectDefinitionLocalService.fetchObjectDefinition(
+				objectDefinitionId);
+
+		if (!objectDefinition.isAccountEntryRestricted()) {
+			return;
+		}
+
+		ObjectField objectField = _objectFieldLocalService.getObjectField(
+			objectDefinition.getAccountEntryRestrictedObjectFieldId());
+
+		long accountEntryId = GetterUtil.getLong(
+			values.get(objectField.getName()));
+
+		if (accountEntryId == 0) {
+			return;
+		}
+
+		long[] accountEntryIds = ListUtil.toLongArray(
+			_accountEntryLocalService.getUserAccountEntries(
+				getUserId(), AccountConstants.PARENT_ACCOUNT_ENTRY_ID_DEFAULT,
+				null,
+				new String[] {
+					AccountConstants.ACCOUNT_ENTRY_TYPE_BUSINESS,
+					AccountConstants.ACCOUNT_ENTRY_TYPE_PERSON
+				},
+				WorkflowConstants.STATUS_APPROVED, QueryUtil.ALL_POS,
+				QueryUtil.ALL_POS),
+			AccountEntry::getAccountEntryId);
+
+		if (!ArrayUtil.contains(accountEntryIds, accountEntryId)) {
+			throw new ObjectDefinitionAccountEntryRestrictedException(
+				StringBundler.concat(
+					"User ", getUserId(),
```

Comment:
@carolmariaabb can we give back a message similar to the permissions framework?

Reply:
@gabrielwas, do you want me to change the exception to `PrincipalException.MustHavePermission` or just the error message?

Reply:
Just changing the message